src/content/resources/javascript/promises/async-await.md

```diff
@@ -0,0 +1,137 @@
+---
+authors:
+  - "ddivad#1337"
+created_at: "2019/10/01"
+title: Async Await
+recommended_reading:
+  - javascript/promises
+---
+
+## Async Await
+
+Async / Await is another way to handle the calling of asynchronous code and is built on top of Promises. It is a nice and clean way to handle asynchronous code, without the need to `.then()` functions to the promise.
+
+It is made up of 2 keywords as the name suggests: `async` and `await`. These need to be used together and can for this method to work.
+
+## Async
+
+The `async` keyword is used to show that the function is going to return a promise. Any return values from the function will be converted to a promise automatically, if they are not already.
+
+```js
+async function foo() {
+  return "foo";
+}
+```
+
+This basic example will return a promise with the value of `foo`.
+
+## Await
+
+The `await` keyword is used to wait until a promise has executed and fetches the result. In order to use the `await` keyword, you **need** to be inside an `async` function. 
+
+```js
+let data = await foo();
+console.log(foo); // "foo"
```

Comment:
This part should probably be explicitly be inside an `async` function, given as we're just introducing the idea for the first time and we wouldn't want any confusion

Reply:
👍